components/file_preview_modal/index.js

```diff
@@ -1,14 +1,16 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import React from 'react';
 import {connect} from 'react-redux';
 
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {makeAsyncComponent} from 'components/async_load';
 
 import {canDownloadFiles} from 'utils/file_utils.jsx';
 
-import FilePreviewModal from './file_preview_modal';
+const FilePreviewModal = makeAsyncComponent(React.lazy(() => import('./file_preview_modal')));
```

Comment:
i am thinking do we still need to do this because of the new ModalController implementation, For FilePreviewModal PR is here https://github.com/mattermost/mattermost-webapp/pull/9334

Reply:
This is not strictly related to that. This make the FilePreviewModal asyncronously loaded, that means all the javascript and dependencies are loaded in another bundle with the first time you render the component. That helps to speed up the initial load by not downloading and parsing javascript code that you are not using yet.

Reply:
Oh i thought It was being lazy loaded on the current implementation that it is always mounted in the component tree.